oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/ClientSecretAuthenticationProvider.java

```diff
@@ -93,6 +95,11 @@ public Authentication authenticate(Authentication authentication) throws Authent
 			throwInvalidClient(OAuth2ParameterNames.CLIENT_ID);
 		}
 
+		Instant expiredAt = registeredClient.getClientSecretExpiresAt();
```

Comment:
Please move this validation below `if (!this.passwordEncoder.matches(clientSecret, registeredClient.getClientSecret()))`